src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java

```diff
@@ -0,0 +1,299 @@
+ * {@code key} metadata property matches the regular expression is determined.
+ * If {@code pattern} is not set, then whether any key named {@code key} is
+ * present determines the match. The key {@code when} determines
+ * whether the {@code decision} is made for matching documents (if that key is
+ * set to {@code found}) or made for non-matching documents (if that key is set
+ * to {@code not-found}). The config key {@code keyset} may be set to
+ * {@code metadata} or to {@code params} to restrict the search to
+ * {@code Metadata} or {@code params}, respectively. If {@code keyset} is not
+ * specified, it defaults to {@code metadata}.
+ * Most keys/values of interest will normally be specified in the document's
+ * {@code Metadata}, but some key/values of interest (e.g. ContentType, DocId)
+ * exist in the document's {@code params}.
+ *
+ * <p>Example: skip documents that have a {@code NoIndex} metadata key or params
+ * key, regardless of value:
+ * <pre><code>
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=NoIndex
+   metadata.transform.pipeline.regexFilter.decision=do-not-index
+   </code></pre>
+ *
+ * <p>Example: drop the content of documents that have a {@code ContentLength}
+ * greater than or equal to 100 megabytes:
+ * <pre><code>
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=ContentLength
+   metadata.transform.pipeline.regexFilter.keyset=params
+   metadata.transform.pipeline.regexFilter.pattern=0*[1-9][0-9]{8,}
+   metadata.transform.pipeline.regexFilter.decision=do-not-index-content
+   </code></pre>
+ *
+ * <p>Example: skips documents whose Metadata {@code Classification} property
+ * is neither {@code PUBLIC} nor {@code DECLASSIFIED}:
+ * <pre><code>
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=Classification
+   metadata.transform.pipeline.regexFilter.keyset=metadata
+   metadata.transform.pipeline.regexFilter.pattern=(PUBLIC)|(DECLASSIFIED)
+   metadata.transform.pipeline.regexFilter.when=not-found
+   metadata.transform.pipeline.regexFilter.decision=do-not-index
+   </code></pre>
+ */
+public class RegexFilter implements MetadataTransform {
+  /**
+   * Make decision based upon whether the regular expression matches or not.
+   */
+  private static enum When {
+    FOUND("found"),
+    NOT_FOUND("not-found");
+
+    private final String name;
+
+    private When(String name) {
+      this.name = name;
+    }
+
+    public static When from(String val) {
+      if ("not-found".equalsIgnoreCase(val)) {
```

Comment:
You could think about using the same technique as Keyset, with replace('-', '_') added.